src/styles/themes/useThemePreference.js

```diff
@@ -0,0 +1,28 @@
+import {useState, useEffect, useContext} from 'react';
+import {Appearance} from 'react-native';
+import CONST from '../../CONST';
+import {PreferredThemeContext} from '../../components/OnyxProvider';
+
+function useThemePreference() {
+    const [themePreference, setThemePreference] = useState(CONST.THEME.DEFAULT);
```

Comment:
Setting a default value for `themePreference` as a state variable and then updating it in `useEffect` caused an unnecessary (and then cancelled) network request here #34669